torch/fx/experimental/proxy_tensor.py

```diff
@@ -137,6 +137,9 @@ def get_proxy_slot(obj, tracer, default=no_default, transform=lambda x: x):
     return transform(tracker[obj])
 
 def snapshot_fake(val):
+    # TODO: skip for sparse to avoid faketensor->tensor change?
+    if val.is_sparse:
+        return val
```

Comment:
This is a straight bug, it's /probably/ OK to fail to snapshot but really detach() should work on fake sparse tensors

Reply:
agreed; I tried to look into the underlying cause, but for a newbee it is not easy to trace the detach() path all the way down to the bug (hopefully @pearu can help). But I am hoping the (temp) bail-out is acceptable, since we never entered this code with sparse anyway before

Reply:
yes, I'll look into the detach bug withing a couple of days, I am just busy with another project atm.

Reply:
heads up: I have a fix for COO detach, once I have one for CSR/... as well, I'll submit the fix.

Reply:
https://github.com/pytorch/pytorch/pull/125679

Reply:
Thanks for fixing that!

Reply:
https://github.com/pytorch/pytorch/pull/125679 has landed, resolving this discussion.